pages/upgrade-guide/4.0.md

```diff
@@ -123,13 +161,17 @@ jQuery 4.0 removes `push`, `sort`, and `splice` from the jQuery prototype. These
 
 The slim build of jQuery 4.0 no longer includes the `callbacks` and `deferred` modules. Use the full build if you need these modules, or use native Promise objects.
 
-#### Breaking change: Remove undocumented `root` parameter of `jQuery()`
+#### Breaking change: Remove undocumented `root` parameter of `jQuery.fn.init()`
 
-The `root` parameter of `jQuery()` was never documented and was only used internally. Also, it has not been necessary since jQuery 1.9 when `jQuery.sub()` was removed.
+`root` was the third parameter of `jQuery.fn.init()`, but was never documented and was only used internally. Also, it has not been necessary since jQuery 1.9 when `jQuery.sub()` was removed.
 
 #### Important fix: Add `exports` to package.json
 
-jQuery exports multiple builds that can be used in different environments. This includes the default build as both a universal module (UMD) and an ESM module, the slim build in UMD and ESM, and a factory build for running jQuery with DOM emulators like JSDOM. The `exports` field in `package.json` is used to specify which build is used in different environments. This change is not expected to affect existing code.
+jQuery exports multiple builds that can be used in different environments. This includes the default build as both a universal module (UMD) and an ESM module, the slim build in UMD and ESM, and a factory build for running jQuery with DOM emulators like JSDOM. The `exports` field in `package.json` is used to specify which build is used in different environments. This change should not affect existing code in most cases, but it is an important fix to ensure that jQuery can be used in a wider range of environments. In fact, jQuery now exports a separate "factory" version for use in Node where the DOM is not available, such as in testing environments like JSDOM. This used to be supported by the main jQuery file, but it is now a separate build and the `exports` field ensures that it is used correctly.
```

Comment:
The factory entry point is mentioned twice, I'll leave it up to you to decide if that's OK.

I was thinking about a more detailed imports usage, but that's perhaps better kept in a blog post and the npm package README.

Reply:
BTW, the `4.0.0-beta.2` README didn't use our fixture but the main README of the jQuery repo:
https://www.npmjs.com/package/jquery/v/4.0.0-beta.2
The fixture was as intended:
https://github.com/jquery/jquery/blob/4.0.0-beta.2/build/fixtures/README.md

This tag already has the new release code, perhaps there's a bug there?

Let's make sure this is using a correct version for the RC.

Reply:
Yeah, the npm package for `4.0.0-beta.2` is off... It contains the `.vscode` folder and a `local.log` file (it looks like it wasn't done on a fresh clone?) and some other files:
https://www.npmjs.com/package/jquery/v/4.0.0-beta.2?activeTab=code

It looks like `npm publish` was done from the `jquery/jquery` repo instead of the proper `jquery/jquery-dist` - that repo has a correct README.

Reply:
I extracted that publishing issue to https://github.com/jquery/jquery/issues/5658.

Reply:
Good catch. I'll make sure the next publish is from the dist repo.

Reply:
I removed the second mention and the explanation about the old way. I think it's okay to keep it simple here. We can go more in-depth in the blog post.